### PR TITLE
Improve `ext.asyncio.engine`

### PR DESCRIPTION
**Pull request by bryanforbes**
### Description
Improve `ext.asyncio.engine`. Depends on #47 and #50.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [X] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: sqlalchemy-stubs/ext/asyncio/engine.pyi [hunk 1]
```diff
@@ -1,103 +1,128 @@
...
-    engine: Any = ...
+    # copied from future.Connection via create_proxy_methods
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def invalidated(self) -> bool: ...
+    dialect: Any
+    @property
+    def default_isolation_level(self) -> Any: ...
+    # end copied
```
Comment (CaselIT) [inline sqlalchemy-stubs/ext/asyncio/engine.pyi:34]: I've also asked here https://github.com/sqlalchemy/sqlalchemy2-stubs/pull/52#discussion_r611037575, can't we create a superclass just in the typing for these?

File: sqlalchemy-stubs/ext/asyncio/engine.pyi [hunk 2]
```diff
@@ -1,103 +1,128 @@
...
+
+    engine: AsyncEngine = ...
     sync_engine: Any = ...
```
Comment (CaselIT) [inline sqlalchemy-stubs/ext/asyncio/engine.pyi:37]: this is actually an Engine iirc

File: sqlalchemy-stubs/ext/asyncio/engine.pyi [hunk 3]
```diff
@@ -1,103 +1,128 @@
...
+    def get_nested_transaction(self) -> Optional[AsyncTransaction]: ...
+    async def execution_options(
+        self: _TAsyncConnection, **opt: Any
+    ) -> _TAsyncConnection: ...
     async def commit(self) -> None: ...
     async def rollback(self) -> None: ...
     async def close(self) -> None: ...
     async def exec_driver_sql(
         self,
         statement: Executable,
```
Comment (CaselIT) [inline sqlalchemy-stubs/ext/asyncio/engine.pyi:67]: this is actually a string

File: sqlalchemy-stubs/ext/asyncio/engine.pyi [hunk 4]
```diff
@@ -1,103 +1,128 @@
...
     async def scalar(
         self,
         statement: Executable,
-        parameters: Optional[Mapping] = ...,
-        execution_options: Mapping = ...,
+        parameters: Optional[Mapping[Any, Any]] = ...,
+        execution_options: Mapping[Any, Any] = ...,
+    ) -> Any: ...
+    async def run_sync(
+        self, fn: Callable[..., Any], *arg: Any, **kw: Any
```
Comment (CaselIT) [inline sqlalchemy-stubs/ext/asyncio/engine.pyi:90]: the first element of the callable is a `Connection`

File: sqlalchemy-stubs/ext/asyncio/engine.pyi [hunk 5]
```diff
@@ -1,103 +1,128 @@
...
     async def __aexit__(
         self, type_: Any, value: Any, traceback: Any
     ) -> None: ...
 
 class AsyncEngine(ProxyComparable, AsyncConnectable):
+    # copied from future.Engine by create_proxy_methods
+    def clear_compiled_cache(self) -> None: ...
+    def update_execution_options(self, **opt: Any) -> None: ...
+    def get_execution_options(self) -> Mapping[Any, Any]: ...
+    # end copied
```
Comment (CaselIT) [inline sqlalchemy-stubs/ext/asyncio/engine.pyi:104]: same wrt superclass

File: sqlalchemy-stubs/ext/asyncio/engine.pyi (hunk 1 above)
Comment (bryanforbes) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:34]: Yeah, I meant to reply there. I think that's a good idea and I'll go ahead and implement that.

Comment (bryanforbes) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:34]: As I said in the other PR: thinking about this a bit more, I think we (me, most likely) should implement this change after all of the current PRs are landed so as to reduce changes between files right now.

Comment (CaselIT) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:34]: ref https://github.com/sqlalchemy/sqlalchemy2-stubs/issues/61

File: sqlalchemy-stubs/ext/asyncio/engine.pyi (hunk 5 above)
Comment (CaselIT) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:104]: https://github.com/sqlalchemy/sqlalchemy2-stubs/issues/61

**Comment by CaselIT** [comment]
I think all these can be fixed later, nothing too important

File: sqlalchemy-stubs/ext/asyncio/engine.pyi (hunk 4 above)
Comment (bryanforbes) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:90]: Unfortunately, I'm not sure we can reliably type that at this time

Comment (CaselIT) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:90]: `Callable[[sometype, ...], ...]` is not a thing I guess

Comment (bryanforbes) [reply sqlalchemy-stubs/ext/asyncio/engine.pyi:90]: No, it's not

**Pull request merged**